### PR TITLE
Webpack[er] support

### PR DESCRIPTION
**Pull request by woahdae**
Previously, ESM modules / sprockets was the only way to include the files in a Rails project.

The official way to include gem-based javascript in a webpacker-based Rails project is to create an npm module, see ActionCable as an example: https://github.com/rails/rails/tree/main/actioncable

This adds webpacker support by copying the ActionCable approach, including using rollup to transpile a plain JS file into ESM and UMD versions.

Related, this also makes the croppable javascript export a function to invoke the croppable functionality instead of hardcoding callbacks on DOMContentLoaded and turbo:load events, which lets the project be more flexible in how it's instantiated (ex. via turbolinks instead of turbo).

### CODE REVIEW

File: app/javascript/croppable/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import Cropper from 'cropperjs';
```
Comment (stevenbarragan) [inline app/javascript/croppable/index.js:1]: Why is this file duplicated? We have`/javascripts/croppable.esm.js` and `app/javascript/croppable/index.js`. Can we just keep one?

File: dist/croppable.css [hunk 2]
```diff
@@ -0,0 +1,71 @@
+.croppable-preview {
```
Comment (stevenbarragan) [inline dist/croppable.css:1]: Why is this file needed? Can we also just keep one `css` file?

**Comment by woahdae** [comment]
Re the duplicate files, I copied the process, including what is committed, from actioncable (https://github.com/rails/rails/tree/main/actioncable). The gist is that they, and thus this commit, works from `app/javascript/croppable/index.js` following the modern file location and naming conventions of javascript in Rails projects, and copies it to match the sprockets conventions. Then the project can serve both as a node package as well as a Rails project. I could, instead, try to figure out how to separate it into two repos, or put everything under `dist` maybe (still not sure why Rails/ActionCable puts things in `app/assets/javascript` instead of `dist` like the css). Would either of these work @stevenbarragan ?

**Comment by woahdae** [comment]
I'm looking to use this in another project, I'll merge this in and in the future if it's a maintenance burden we can figure it out.
